tcms/testplans/tests/test_edit_plan.py

```diff
@@ -0,0 +1,58 @@
+from http import HTTPStatus
+
+from django.urls import reverse
+
+from tcms.tests import user_should_have_perm, BasePlanCase
+
+
+class EditPlanViewTest(BasePlanCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.testplan_edit_url = reverse('plan-edit', args=[cls.plan.pk])
+
+        cls.testplan_edit_data = {
+            'author': cls.tester.pk,
+            'product': cls.plan.product.pk,
+            'product_version': cls.plan.product_version.pk,
+            'type': cls.plan.type.pk,
+            'name': cls.plan.name,
+
+            'email_settings-0-auto_to_plan_author': 'on',
+            'email_settings-0-auto_to_case_owner': 'on',
+            'email_settings-0-auto_to_case_default_tester': 'on',
+            'email_settings-0-notify_on_case_update': 'on',
+            'email_settings-0-notify_on_plan_update': 'on',
+
+            'email_settings-0-plan': '',
+            'email_settings-0-id': cls.plan.emailing.pk,
+            'email_settings-TOTAL_FORMS': '1',
+            'email_settings-INITIAL_FORMS': '1',
+            'email_settings-MIN_NUM_FORMS': '0',
+            'email_settings-MAX_NUM_FORMS': '1',
+        }
+
+        user_should_have_perm(cls.tester, perm='testplans.change_testplan')
+
+    def test_404_if_testplan_id_not_exist(self):
+        url = reverse('plan-edit', args=[99999])
+        response = self.client.get(url)
+        self.assertEqual(HTTPStatus.NOT_FOUND, response.status_code)
```

Comment:
This functionality is provided by the underlying UpdateView and we don't have to test it explicitly.